src/main/java/org/zalando/fauxpas/FauxPas.java

```diff
@@ -56,4 +60,22 @@
         return predicate;
     }
 
+    public static <R> Function<Throwable, R> partially(final ThrowingFunction<Throwable, R, Throwable> function) {
+        return throwable -> {
+            try {
+                return function.tryApply(unpack(throwable));
+            } catch (final CompletionException e) {
+                throw e;
+            } catch (final Throwable e) {
+                throw new CompletionException(e);
+            }
+        };
+    }
+
+    private static Throwable unpack(final Throwable throwable) {
+        final boolean isCompletionException = throwable instanceof CompletionException;
+        final boolean hasCause = throwable.getCause() != null;
+        return isCompletionException && hasCause ? throwable.getCause() : throwable;
```

Comment:
Also docs should be updated with the fact that function will receive either cause or CompletionException if thzere is no cause